thirdparty/mbedtls/include/mbedtls/aesni.h

```diff
@@ -54,9 +54,10 @@
  *       macros that may change in future releases.
  */
 #undef MBEDTLS_AESNI_HAVE_INTRINSICS
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && (defined(_M_AMD64) || defined(_M_IX86))
 /* Visual Studio supports AESNI intrinsics since VS 2008 SP1. We only support
- * VS 2013 and up for other reasons anyway, so no need to check the version. */
+ * VS 2013 and up for other reasons anyway, so no need to check the version.
+ * Only supported on x64 and x86. */
```

Comment:
Small note, could also be "Only supported on x86 architectures (x86_64 or x86_32)" or similar. But this is also good as-is in the context of MSVC, since Microsoft uses the term "x64" to refer to x86_64.

Reply:
Yeah I chose to use the Microsoft terminology here.